### DISCUSSION TITLE
Svelte 5 Snippets with renderComponent

### DISCUSSION THREAD

**Discussion by jacokok** [TanStack/table#5742] [Ideas]
Svelte 5 snippets is working great for headers and footers in tables.
Can we also use them for cell values?

```ts
const columns: ColumnDef<DataType>[] = [
	{
		accessorKey: "id",
		cell: (info) => info.getValue(),
	},
	{
		accessorKey: "name",
		cell: (info) => renderComponent(custom),
	},
];
```

```svelte
{#snippet custom()}
	<Badge variant="outline">Test</Badge>
{/snippet}
```

The above code works and renders the snippet with typescript complaining.
Could we possibly allow snippets as the arguments to renderComponent or create another method to render snippets?

**Comment by jacokok** [comment]
```svelte
const columns: ColumnDef<DataType>[] = [
	{
		accessorKey: "name",
		cell: (info) => custom(),
	},
];
```

I am guessing it should actually look like this. Just run a function but I am getting a runtime error running it like this.

**Comment by stefa168** [comment]
How would this work with `renderComponent` and a snippet that expects a prop? I'm currently having some issues with them on Svelte 5.

**Reply by jacokok** [reply]
This now works as expected. Check out dataTable utils from svelte shadcn
